var/spack/repos/builtin/packages/py-merlin/package.py

```diff
@@ -23,14 +24,20 @@ class PyMerlin(PythonPackage):
     depends_on("python@3.6:", type=("build", "run"))
     depends_on("py-setuptools", type=("build", "run"))
     depends_on("py-cached-property", type=("build", "run"))
-    depends_on("py-celery@5.0.0+redis+sqlalchemy", when="@1.7.5:", type=("build", "run"))
-    depends_on("py-celery@4.3.0:4+redis+sqlalchemy", when="@:1.7.4", type=("build", "run"))
-    depends_on("py-coloredlogs@10.0:", type=("build", "run"))
+    depends_on("py-celery@4.3.0:+redis+sqlalchemy", type=("build", "run"))
+    depends_on("py-celery@5.0.3:", when="@1.7.6:", type=("build", "run"))
+    depends_on("py-celery@5.0.0:", when="@1.7.5", type=("build", "run"))
+    depends_on("py-celery@4.4.5:", when="@1.6.2:", type=("build", "run"))
+    depends_on("py-celery@:4", when="@:1.7.4", type=("build", "run"))
```

Comment:
Writing it this way makes understanding what is going on not easy (sorry, just my opinion). I also checked the sources for 1.4.1 (not in git, only on pypi) and we missed changes there too.
```suggestion
    depends_on("py-celery@5.0.3:+redis+sqlalchemy", when="@1.7.6:", type=("build", "run"))
    depends_on("py-celery@5.0.0:+redis+sqlalchemy", when="@1.7.5", type=("build", "run"))
    depends_on("py-celery@4.4.5:+redis+sqlalchemy", when="@1.6.2:1.7.4", type=("build", "run"))
    depends_on("py-celery@4.3.0:+redis+sqlalchemy", when="@:1.5.2", type=("build", "run"))
    depends_on("py-celery@4.3.0:+redis", when="@1.4.1:1.5.1", type=("build", "run"))

```

Reply:
Again, we need that upperbound `@:4` whether it's on mypy or not. It's a common theme that the python community doesn't set upperbound and then are surprised when a new major release "breaks" their package.

Reply:
I have nothing against the upper bound. If you say it does not work with newer ´py-celery` versions, can you add a comment why this is needed so that we don't have to guess for future reviews?

As to how this is added, I have no preference.

Reply:
Thanks.